src/combined-selectors/reporting-config-search-results.ts

```diff
@@ -54,16 +55,100 @@ function searchReportingConfig(
+			}
+			scores[ id ] += weight;
+		}
+	}
+
+	const scoreThreshold = 1;
+
+	for ( const entityId in scores ) {
+		if ( scores[ entityId ] >= scoreThreshold ) {
+			if ( products[ entityId ] ) {
+				searchResults.products.add( entityId );
+			} else if ( featureGroups[ entityId ] ) {
+				searchResults.featureGroups.add( entityId );
+				searchResults.products.add( featureGroups[ entityId ].productId );
+			} else if ( features[ entityId ] ) {
+				searchResults.features.add( entityId );
+				if ( features[ entityId ].parentType === 'product' ) {
+					searchResults.products.add( features[ entityId ].parentId );
+				} else {
+					searchResults.featureGroups.add( features[ entityId ].parentId );
+					const parentFeatureGroup = featureGroups[ features[ entityId ].parentId ];
+					searchResults.products.add( parentFeatureGroup.productId );
+				}
+			}
+		}
+	}
+
 	return searchResults;
 }
 
+function addTokensToInvertedIndex(
+	entityId: string,
+	description: string | undefined,
+	invertedIndex: Map< string, Array< { id: string; weight: number } > >
+) {
+	const descriptionTokens = description ? tokenizeAndNormalize( description ) : [];
+	const tokensWithWeights = [ ...descriptionTokens.map( ( token ) => ( { token, weight: 1 } ) ) ];
+
+	for ( const { token, weight } of tokensWithWeights ) {
+		if ( ! invertedIndex.has( token ) ) {
+			invertedIndex.set( token, [] );
+		}
+
+		const tokenList = invertedIndex.get( token );
+		if ( tokenList ) {
+			tokenList.push( { id: entityId, weight } );
+		}
+	}
+}
+
+function createDescriptionInvertedIndex( reportingConfig: NormalizedReportingConfig ) {
+	const invertedIndex = new Map();
+
+	const { features, featureGroups, products } = reportingConfig;
+
+	for ( const productId in products ) {
+		const product = products[ productId ];
+		addTokensToInvertedIndex( productId, product.description, invertedIndex );
+	}
+
+	for ( const featureGroupId in featureGroups ) {
+		const featureGroup = featureGroups[ featureGroupId ];
+		addTokensToInvertedIndex( featureGroupId, featureGroup.description, invertedIndex );
+	}
+
+	for ( const featureId in features ) {
+		const feature = features[ featureId ];
+		addTokensToInvertedIndex( featureId, feature.description, invertedIndex );
+	}
+
+	return invertedIndex;
+}
+
 // Searching all of the reporting config is expensive, and these search results are needed by several components.
 // For performance, we need to memo-ize (cache) this piece of derived state.
 // The "createSelector" function lets you  do just that:
 // https://redux.js.org/usage/deriving-data-selectors#optimizing-selectors-with-memoization
 export const selectReportingConfigSearchResults = createSelector(
 	[ selectFeatureSearchTerm, selectNormalizedReportingConfig ],
 	( searchTerm, reportingConfig ) => {
-		return searchReportingConfig( searchTerm, reportingConfig );
+		return searchReportingConfig(
+			searchTerm,
+			reportingConfig,
+			createDescriptionInvertedIndex( reportingConfig )
```

Comment:
**Suggestion (blocking):** Ah, I actually think this may not cache exactly how we need it to. The question of where to do the caching though is kind of a tricky one, so I'll spin up a Slack thread to discuss if that's okay with you! 😄

Reply:
Slack thread: p1682459226372729-slack-CQD1HH4MA